generated/nidcpower/nidcpower_service.cpp

```diff
@@ -2168,7 +2168,30 @@ namespace nidcpower_grpc {
       return ::grpc::Status::CANCELLED;
     }
     try {
-      return ::grpc::Status(::grpc::UNIMPLEMENTED, "TODO: This server handler has not been implemented.");
+      auto vi_grpc_session = request->vi();
+      ViSession vi = session_repository_->access_session(vi_grpc_session.id(), vi_grpc_session.name());
+      ViConstString channel_name = request->channel_name().c_str();
+      ViReal64 timeout = request->timeout();
+      ViInt32 count = request->count();
+      response->mutable_voltage_measurements()->Resize(count, 0);
+      ViReal64* voltage_measurements = response->mutable_voltage_measurements()->mutable_data();
+      response->mutable_current_measurements()->Resize(count, 0);
+      ViReal64* current_measurements = response->mutable_current_measurements()->mutable_data();
+      std::vector<ViBoolean> in_compliance(count, ViBoolean());
+      response->mutable_in_compliance()->Resize(count, 0);
+      auto in_compliance_mutable_data = response->mutable_in_compliance()->mutable_data();
+      ViInt32 actual_count {};
+      auto status = library_->FetchMultiple(vi, channel_name, timeout, count, voltage_measurements, current_measurements, in_compliance.data(), &actual_count);
+      response->set_status(status);
+      if (status == 0) {
+        int i = 0;
```

Comment:
You would get errors with the `i` variable being defined multiple times if there were two `ViBoolean[]` output parameters. For each custom type we generate `Copy` functions at the top of the service that take in a vector and a `RepeatedPtrField`. Instead of using `mutable_data()`, we simply call `AddAllocated`. I think the better approach would be to follow a similar pattern. The only difference would be that we call `Add` instead of `AddAllocated`. Then this for loop turns into this: `Copy(in_compliance, response->mutable_in_compliance()`.

Reply:
I agree, I think following the `Copy` pattern done for custom structs seems like the best option for handling the bool to ViBoolean array conversions. So having the appropriate `Copy` methods defined in `service.h.mako` and `service.cpp.mako` for `ViBoolean` to `bool` should do the trick and maintain that pattern.

Reply:
The reason why defining `Copy` function for ViBoolean to bool array conversion (or other way around) is tricky, is because we will need to look at parameters for all functions to see if any of them are `ViBoolean[]` in service.h.mako to determine if we need to define a Copy function. For structs, we just iterate over each entry in `custom_types` and define Copy function for each of them.

Reply:
I wouldn't call enumerating through all function parameters as tricky. When determining which enums to include in the proto file we enumerate through all attributes and function parameters for enum references. The build still performs well. Even if it took an entire second to perform the calculation (which it wouldn't), adding 1 sec to the build time is unnoticeable.

Reply:
Added `Copy` function in `service.h.mako` and `service.cpp.mako`. The `Copy` function is pretty much tied (hardcoded) to `ViBoolean` and `bool` though. See my comment on the `service.cpp.mako`